migration-guide.md

```diff
@@ -0,0 +1,500 @@
+# Migration guide
+
+The purpose of this guide is to help easily upgrade to Azure Cosmos DB Java SDK 4.0 for Core (SQL) API ("Java SDK 4.0" from here on out.) The audience for this guide is current users of
+
+* "Legacy" Sync Java SDK 2.x.x
+* Async Java SDK 2.x.x
+* Java SDK 3.x.x
+
+## Background
+
+| Java SDK                | Release Date | Bundled APIs         | Maven Jar                               | Java package name                |API Reference                                              | Release Notes                                                                            |
+|-------------------------|--------------|----------------------|-----------------------------------------|----------------------------------|-----------------------------------------------------------|------------------------------------------------------------------------------------------|
+| Async 2.x.x             | June 2018    | Async(RxJava)        | com.microsoft.azure::azure-cosmosdb     | com.microsoft.azure.cosmosdb.rx  | [API](https://azure.github.io/azure-cosmosdb-java/2.0.0/) | [Release Notes](https://docs.microsoft.com/en-us/azure/cosmos-db/sql-api-sdk-async-java) |
+| "Legacy" Sync 2.x.x     | Sept 2018    | Sync                 | com.microsoft.azure::azure-documentdb   | com.microsoft.azure.cosmosdb     | [API](https://azure.github.io/azure-cosmosdb-java/2.0.0/) | [Release Notes](https://docs.microsoft.com/en-us/azure/cosmos-db/sql-api-sdk-java)       |
```

Comment:
added some comments on this PR: 
https://github.com/Azure-Samples/azure-cosmos-java-sql-api-samples/pull/7#pullrequestreview-384242924

please address this ones. 
for new comments I will add in this PR.